### PR TITLE
feature: add methods for navigating tree structure of data

### PR DESCRIPTION
**Pull request by schwager-hsph**
Hi Victor, I added a few convenience methods for navigating the data tree. 
I wasn't sure how to test the methods meaningfully; if you'd like for me to write tests for these methods, just give me a little direction and I'll get to it.

### CODE REVIEW

File: cutlass/Project.py [hunk 1]
```diff
@@ -399,3 +401,18 @@ def _get_raw_doc(self):
             project_doc['ver'] = self._version
 
         return project_doc
+
+
+    def studies(self):
+        """Return iterator of all studies part of this project """
+        q = '"{}"[linkage.part_of]'.format(self.id)
```
Comment (victor73) [inline cutlass/Project.py]: Can you rename "q" to something else?

File: cutlass/Sample.py [hunk 2]
```diff
@@ -403,3 +406,40 @@ def _get_raw_doc(self):
             sample_doc['meta']['supersite'] = self._supersite
 
         return sample_doc
+
+
+    def _prep_docs(self):
+        q = '"{}"[linkage.prepared_from]'.format(self.id)
```
Comment (victor73) [inline cutlass/Sample.py]: Can you rename "q" to something else?

File: cutlass/SixteenSDnaPrep.py [hunk 3]
```diff
@@ -729,3 +731,18 @@ def save(self):
                                   __name__ + " %s. Reason: %s" % self._d, e)
 
         return success
+
+
+    def raw_seq_sets(self):
+        """Return iterator of all raw_seq_sets sequenced from this prep """
+        q = '"{}"[linkage.sequenced_from]'.format(self.id)
```
Comment (victor73) [inline cutlass/SixteenSDnaPrep.py]: Can you rename "q" to something else?

File: cutlass/SixteenSRawSeqSet.py [hunk 4]
```diff
@@ -786,3 +789,20 @@ def save(self):
...
 
         return success
+
+
+    def trimmed_seq_sets(self):
+        """Return iterator of all trimmed seq sets that were computed from
+        this seq set
+        
+        """
+        q = '"{}"[linkage.computed_from]'.format(self.id)
```
Comment (victor73) [inline cutlass/SixteenSRawSeqSet.py]: Can you rename "q" to something else?

File: cutlass/Study.py [hunk 5]
```diff
@@ -521,4 +523,32 @@ def is_valid(self):
         
         self.logger.debug("Valid? %s" + str(valid))
 
-        return valid
\ No newline at end of file
+        return valid
+
+    def studies(self):
+        """Return iterator of all studies that are subsets of this study """
+        q = '"{}"[linkage.subset_of]'.format(self.id)
```
Comment (victor73) [inline cutlass/Study.py]: Can you rename "q" to something else?

File: cutlass/Study.py [hunk 6]
```diff
@@ -521,4 +523,32 @@ def is_valid(self):
...
+            for doc in res['results']:
+                yield Study.load_study(doc)
+            res_count -= len(res['results'])
+            if res_count < 1:
+                break
+
+
+    def subjects(self):
+        """Return iterator of all subjects that participate in this study"""
+        q = '"{}"[linkage.participates_in]'.format(self.id)
```
Comment (victor73) [inline cutlass/Study.py]: Can you rename "q" to something else?

File: cutlass/Subject.py [hunk 7]
```diff
@@ -482,3 +484,17 @@ def save(self):
                 self.logger.error("Unable to update " + __name__ + ". Reason: %s" % e.strerror)
 
         return success
+
+
+    def visits(self):
+        """Return iterator of all visits by this subject """
+        q = '"{}"[linkage.by]'.format(self.id)
```
Comment (victor73) [inline cutlass/Subject.py]: Can you rename "q" to something else?

File: cutlass/Visit.py [hunk 8]
```diff
@@ -535,3 +537,18 @@ def save(self):
                                   "Reason: %s" % self._id, e)
 
         return success
+
+
+    def samples(self):
+        """Return iterator of all samples collected during this visit """
+        q = '"{}"[linkage.collected_during]'.format(self.id)
```
Comment (victor73) [inline cutlass/Visit.py]: Can you rename "q" to something else?

File: cutlass/WgsDnaPrep.py [hunk 9]
```diff
@@ -666,3 +668,19 @@ def save(self):
                                   __name__ + " %s. Reason: %s" % self._d, e)
 
         return success
+
+
+    def raw_seq_sets(self):
+        """Return iterator of all raw_seq_sets sequenced from this prep """
+        q = '"{}"[linkage.sequenced_from]'.format(self.id)
```
Comment (victor73) [inline cutlass/WgsDnaPrep.py]: Can you rename "q" to something else?

**Comment by victor73** [comment]
This PR looks great, just a a minor variable naming quibble and we can merge...

**Comment by schwager-hsph** [comment]
Thoughts?

**Pull request merged**